src/java/org/apache/cassandra/index/sai/memory/MSBRadixSorter.java

```diff
@@ -0,0 +1,294 @@
+/*
```

Comment:
The formatting is different than this project requires.
Is there any reason we had to make a copy-paste of the source code from lucene instead of referencing a jar? 

@mike-tr-adamson do we allow non-standard formatting for files copied from external sources? Or should we reformat it to meet the code-style guidelines of the project? Reformatting could have a potential downside that applying changes from lucene might be harder (and maybe even not possible automatically). But on the other hand, it looks weird.

Reply:
private -> ```protected final int maxLength;```

Reply:
I added byte block pooling to avoid the allocation of numerous `BytesRef` objects.  Lucene classes have been imported due to various method visibility restrictions.